pandas/core/indexers.py

```diff
@@ -376,6 +376,27 @@ def unpack_1tuple(tup):
     return tup
 
 
+def check_key_length(columns, key, value):
```

Comment:
can you type as much as you can here

Reply:
We would have to type wiht Index and DataFrame causing circular imports. Is there a way around this?

Reply:
Sorry, should have said that I have already tried this. If using TYPE_CHECKING and typing the signature (not the return value) this raises
NameError: name 'DataFrame' is not defined

Reply:
Did not know about the future import. Thx for the help